src/Bundle/Resources/config/services/controller.xml

```diff
@@ -19,40 +19,69 @@
             <argument type="service" id="service_container" />
             <argument type="service" id="sylius.expression_language" />
         </service>
+        <service id="Sylius\Bundle\ResourceBundle\Controller\ParametersParserInterface" alias="sylius.resource_controller.parameters_parser" public="false" />
+
         <service id="sylius.resource_controller.request_configuration_factory" class="Sylius\Bundle\ResourceBundle\Controller\RequestConfigurationFactory" public="false">
             <argument type="service" id="sylius.resource_controller.parameters_parser" />
             <argument>Sylius\Bundle\ResourceBundle\Controller\RequestConfiguration</argument>
             <argument>%sylius.resource.settings%</argument>
         </service>
+        <service id="Sylius\Bundle\ResourceBundle\Controller\RequestConfigurationFactoryInterface" alias="sylius.resource_controller.request_configuration_factory" public="false" />
+
         <service id="sylius.resource_controller.new_resource_factory" class="Sylius\Bundle\ResourceBundle\Controller\NewResourceFactory" public="false" />
+        <service id="Sylius\Bundle\ResourceBundle\Controller\NewResourceFactoryInterface" alias="sylius.resource_controller.new_resource_factory" public="false" />
+
         <service id="sylius.resource_controller.single_resource_provider" class="Sylius\Bundle\ResourceBundle\Controller\SingleResourceProvider" public="false" />
-        <service id="sylius.resource_controller.pagerfanta_representation_factory" class="Hateoas\Representation\Factory\PagerfantaFactory" public="false" />
+        <service id="Sylius\Bundle\ResourceBundle\Controller\SingleResourceProviderInterface" alias="sylius.resource_controller.single_resource_provider" public="false" />
+
+        <service id="Hateoas\Representation\Factory\PagerfantaFactory" public="false" />
+        <service id="sylius.resource_controller.pagerfanta_representation_factory" alias="Hateoas\Representation\Factory\PagerfantaFactory" public="false" />
+
```

Comment:
I am wondering. For consistency reasons, I'd rather go with named service, then alias instead of service, then named alias.

WDYT?